### PR TITLE
fix(smart-action-hook): value injected to an enum field of type is now correctly handled

### PR DESCRIPTION
**Pull request by SteveBunlon**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Create automatic tests
- [x] No automatic tests failures
- [ ] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: spec/requests/actions_controller_spec.rb [hunk 1]
```diff
@@ -84,7 +89,7 @@
     }
     enums_action_definition = {
       name: 'enums_action',
-      fields: [foo, enum],
+      fields: [foo, enum, multiple_enum],
```
Comment (GuillaumeCisco) [inline spec/requests/actions_controller_spec.rb:92]: Why is `multiple_enum` in `enums_action_definition`?
Can we keep it separated?

`enums_action_definition` with `fields: [foo, enum],`
and
`multiple_enums_action_definition` with `fields: [foo, multiple_enum],`

Would be easier to read and review :)

**Review by GuillaumeCisco — COMMENTED** [review]
Great work! 🦖 
I absolutely love the `-` between arrays for intersection in ruby ❤️ 

I added a comment for dealing with tests ;)

File: spec/requests/actions_controller_spec.rb (hunk 1 above)
Comment (SteveBunlon) [reply spec/requests/actions_controller_spec.rb:92]: True ! I'm changing this

**Review by GuillaumeCisco — APPROVED** [review]
That's perfect now! ✨

**Pull request merged**